src/server/IoDispatchers.cpp

```diff
@@ -269,6 +269,118 @@ static bool _shouldAttemptHandoff(const Globals& globals,
 #endif
 }
 
+static void attemptHandoff(Globals& Globals, const CONSOLE_INFORMATION& gci, CONSOLE_API_CONNECTINFO& cac, PCONSOLE_API_MSG pReceiveMsg)
+{
+    if (!_shouldAttemptHandoff(Globals, gci, cac))
+    {
+        return;
+    }
+
+    try
+    {
+        // Go get ourselves some COM.
+        auto coinit = wil::CoInitializeEx(COINIT_MULTITHREADED);
+
+        // Get the class/interface to the handoff handler. Local machine only.
+        ::Microsoft::WRL::ComPtr<IConsoleHandoff> handoff;
+        THROW_IF_FAILED(CoCreateInstance(Globals.delegationPair.console, nullptr, CLSCTX_LOCAL_SERVER, IID_PPV_ARGS(&handoff)));
+
+        if (Globals.defaultTerminalMarkerCheckRequired)
+        {
+            ::Microsoft::WRL::ComPtr<IDefaultTerminalMarker> marker;
+            if (FAILED(handoff.As(&marker)))
+            {
+                Globals.delegationPair = DelegationConfig::ConhostDelegationPair;

```

Comment:
Will this result in the propsheet showing "Windows Console Host"? Or will it show "Let Windows Decide" still?

Reply:
oh, it will still show LWD because this code isn't used by propsheet

Reply:
The propsheet calls `s_GetAvailablePackages` itself again and doesn't depend on `Globals`.